packages/xo-server-sdn-controller/src/index.js

```diff
@@ -103,13 +103,25 @@ function updateNetworkOtherConfig(network) {
   )
 }
 
+// -----------------------------------------------------------------------------
+
+function createPassword() {
+  const chars = [
+    ...'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789?!',
+  ]
+
+  return [...Array(16)].map(i => chars[(Math.random() * chars.length) | 0])
+    .join``
```

Comment:
This is not a fair random, use [`_.sample`](https://devdocs.io/lodash~4/index#sample).

Use [`Array.from`](https://devdocs.io/javascript/global_objects/array/from) for a simpler and more efficient implementation (you are allocating 3 arrays here):

```js
return Array.from({ length: 16 }, _ => sample(chars))
```